// unipile-node-sdk/dist/esm/messaging/chat-attendees/ressource.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatAttendeeSchema = exports.ChatAttendeeLinkedinSpecifics = void 0;
const typebox_1 = require("@sinclair/typebox");
const common_types_js_1 = require("../../common/common.types.js");
/**
 *
 */
exports.ChatAttendeeLinkedinSpecifics = typebox_1.Type.Object({
    provider: typebox_1.Type.Literal("LINKEDIN"),
    member_urn: typebox_1.Type.String(),
    occupation: typebox_1.Type.Optional(typebox_1.Type.String()),
    network_distance: typebox_1.Type.Optional(typebox_1.Type.Union([
        typebox_1.Type.Literal("DISTANCE_1"),
        typebox_1.Type.Literal("DISTANCE_2"),
        typebox_1.Type.Literal("DISTANCE_3"),
        typebox_1.Type.Literal("OUT_OF_NETWORK"),
    ])),
    pending_invitation: typebox_1.Type.Optional(typebox_1.Type.Boolean()),
    location: typebox_1.Type.Optional(typebox_1.Type.String()),
    headline: typebox_1.Type.Optional(typebox_1.Type.String()),
    contact_info: typebox_1.Type.Optional(typebox_1.Type.Object({
        emails: typebox_1.Type.Optional(typebox_1.Type.Array(typebox_1.Type.String())),
        phone_numbers: typebox_1.Type.Optional(typebox_1.Type.Array(typebox_1.Type.String())),
        websites: typebox_1.Type.Optional(typebox_1.Type.Array(typebox_1.Type.String())),
        social_handles: typebox_1.Type.Optional(typebox_1.Type.Array(typebox_1.Type.Object({ type: typebox_1.Type.String(), name: typebox_1.Type.String() }))),
    })),
});
/**
 *
 */
exports.ChatAttendeeSchema = typebox_1.Type.Object({
    id: common_types_js_1.UniqueIdSchema,
    account_id: common_types_js_1.UniqueIdSchema,
    provider_id: typebox_1.Type.String(),
    name: typebox_1.Type.String(),
    is_self: typebox_1.Type.Union([typebox_1.Type.Literal(1), typebox_1.Type.Literal(0)]),
    hidden: typebox_1.Type.Optional(typebox_1.Type.Union([typebox_1.Type.Literal(1), typebox_1.Type.Literal(0)])),
    picture_url: typebox_1.Type.Optional(typebox_1.Type.String()),
    profile_url: typebox_1.Type.Optional(typebox_1.Type.String()),
    specifics: typebox_1.Type.Optional(typebox_1.Type.Union([exports.ChatAttendeeLinkedinSpecifics], {
        description: "Provider specific additional data.",
    })),
});
