// unipile-node-sdk/dist/cjs/mails/mail.update.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailUpdatedApiResponseValidator = exports.MailUpdatedApiResponseSchema = exports.MailUpdateQueryValidator = exports.MailUpdateQuerySchema = exports.MailUpdateBodyValidator = exports.MailUpdateBodySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const query_parameters_type_js_1 = require("../common/query-parameters.type.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
/**
 *
 */
exports.MailUpdateBodySchema = typebox_1.Type.Object({
    unread: typebox_1.Type.Optional(typebox_1.Type.Boolean({ description: "Indicates whether the email should be marked as unread." })),
    folders: typebox_1.Type.Optional(typebox_1.Type.Array(typebox_1.Type.String(), { description: "The names of the folders to which the email should be moved. Outlook and IMAP accounts accept only one folder." })),
});
exports.MailUpdateBodyValidator = compiler_1.TypeCompiler.Compile(exports.MailUpdateBodySchema);
// export const getMailUpdateBodyOpenApiSchema = makeOpenApiSchemaGetter(MailUpdateBodySchema);
/**
 *
 */
const MailUpdateOptionsQuerySchema = typebox_1.Type.Object({
    account_id: typebox_1.Type.Optional(query_parameters_type_js_1.AccountIdParamSchema),
});
/**
 *
 */
exports.MailUpdateQuerySchema = MailUpdateOptionsQuerySchema;
/**
 *
 */
exports.MailUpdateQueryValidator = compiler_1.TypeCompiler.Compile(exports.MailUpdateQuerySchema);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
/**
 *
 */
exports.MailUpdatedApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("EmailUpdated"),
}, { description: "@todo Email has been updated." });
/**  */
exports.MailUpdatedApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.MailUpdatedApiResponseSchema);
