// unipile-node-sdk/dist/cjs/schemas/chat-attendee.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAllChatAttendeesResponseSchema = exports.getAttendeeResponseSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const chatAttendeeSchema = typebox_1.Type.Object({
    id: typebox_1.Type.String(),
    account_id: typebox_1.Type.String(),
    provider_id: typebox_1.Type.String(),
    name: typebox_1.Type.String(),
    is_self: typebox_1.Type.Union([typebox_1.Type.Literal(1), typebox_1.Type.Literal(0)]),
    hidden: typebox_1.Type.Optional(typebox_1.Type.Union([typebox_1.Type.Literal(1), typebox_1.Type.Literal(0)])),
    picture_url: typebox_1.Type.Optional(typebox_1.Type.String()),
    profile_url: typebox_1.Type.Optional(typebox_1.Type.String()),
});
exports.getAttendeeResponseSchema = typebox_1.Type.Composite([
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal('ChatAttendee'),
    }),
    chatAttendeeSchema,
]);
exports.getAllChatAttendeesResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('ChatAttendeeList'),
    items: typebox_1.Type.Array(exports.getAttendeeResponseSchema),
});
