// unipile-node-sdk/dist/cjs/schemas/chat.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postChatMessageResponseSchema = exports.getChatAttendeesResponseSchema = exports.getChatMessagesResponseSchema = exports.getChatResponseSchema = exports.getChatsResponseSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const chat_attendee_schema_js_1 = require("./chat-attendee.schema.js");
const message_schema_js_1 = require("./message.schema.js");
const utils_schema_js_1 = require("./utils.schema.js");
const ChatFeaturesSchema = typebox_1.Type.Union([typebox_1.Type.Literal('reactions')]);
const chatSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('Chat'),
    id: typebox_1.Type.String(),
    account_id: typebox_1.Type.String(),
    account_type: typebox_1.Type.String(),
    provider_id: typebox_1.Type.String(),
    attendee_provider_id: typebox_1.Type.Optional(typebox_1.Type.String()),
    name: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
    type: typebox_1.Type.Enum({ SINGLE: 0, GROUP: 1, CHANNEL: 2 }),
    timestamp: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
    unread_count: typebox_1.Type.Number(),
    archived: typebox_1.Type.Union([typebox_1.Type.Literal(0), typebox_1.Type.Literal(1)]),
    muted_until: typebox_1.Type.Union([typebox_1.Type.Literal(-1), typebox_1.Type.String(), typebox_1.Type.Null()]),
    read_only: typebox_1.Type.Union([typebox_1.Type.Literal(0), typebox_1.Type.Literal(1), typebox_1.Type.Literal(2)]),
    disabledFeatures: typebox_1.Type.Optional(typebox_1.Type.Array(ChatFeaturesSchema)),
    lastMessage: typebox_1.Type.Optional(typebox_1.Type.Union([message_schema_js_1.messageSchema, typebox_1.Type.Null()])),
});
exports.getChatsResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('ChatList'),
    items: typebox_1.Type.Array(chatSchema),
    cursor: utils_schema_js_1.cursorSchema,
});
exports.getChatResponseSchema = typebox_1.Type.Composite([
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal('Chat'),
    }),
    chatSchema,
]);
exports.getChatMessagesResponseSchema = message_schema_js_1.getMessagesResponseSchema;
exports.getChatAttendeesResponseSchema = chat_attendee_schema_js_1.getAllChatAttendeesResponseSchema;
exports.postChatMessageResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('MessageSent'),
});
