// unipile-node-sdk/dist/cjs/schemas/utils.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UTCDateTimeMsTypeSchema = exports.accountSourceProviderSchema = exports.messagingProviderSchema = exports.listCursorQuerySchema = exports.listLimitQuerySchema = exports.cursorSchema = exports.uniqueIdSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
exports.uniqueIdSchema = typebox_1.Type.String();
exports.cursorSchema = typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]);
exports.listLimitQuerySchema = typebox_1.Type.Object({
    limit: typebox_1.Type.Optional(typebox_1.Type.Integer({
        minimum: 1,
        maximum: 250,
    })),
});
exports.listCursorQuerySchema = typebox_1.Type.Composite([typebox_1.Type.Object({ cursor: exports.cursorSchema }), exports.listLimitQuerySchema]);
exports.messagingProviderSchema = typebox_1.Type.Union([
    typebox_1.Type.Literal('WHATSAPP'),
    typebox_1.Type.Literal('LINKEDIN'),
    typebox_1.Type.Literal('SLACK'),
    typebox_1.Type.Literal('TWITTER'),
    typebox_1.Type.Literal('INSTAGRAM'),
    typebox_1.Type.Literal('MESSENGER'),
    typebox_1.Type.Literal('TELEGRAM'),
]);
exports.accountSourceProviderSchema = typebox_1.Type.Union([
    typebox_1.Type.Literal('GOOGLE'),
    typebox_1.Type.Literal('GOOGLE_CALENDAR'),
    typebox_1.Type.Literal('ICLOUD'),
    typebox_1.Type.Literal('LINKEDIN'),
    typebox_1.Type.Literal('MAIL'),
    typebox_1.Type.Literal('MOBILE'),
    typebox_1.Type.Literal('OUTLOOK'),
    typebox_1.Type.Literal('TWITTER'),
    typebox_1.Type.Literal('WHATSAPP'),
    typebox_1.Type.Literal('SLACK'),
]);
exports.UTCDateTimeMsTypeSchema = typebox_1.Type.String();
