// unipile-node-sdk/dist/cjs/webhooks/webhooks-create.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookCreateResponseValidator = exports.WebhookCreateResponseSchema = exports.WebhookCreateBodyValidator = exports.WebhookCreateBodySchema = exports.WebhookCreateAccountsStatusBodySchema = exports.WebhookCreateEmailTrackingBodySchema = exports.WebhookCreateEmailBodySchema = exports.WebhookCreateMessagingBodySchema = exports.WebhookCreateBaseBodySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const ressource_types_js_1 = require("./ressource.types.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
/**
 *
 */
exports.WebhookCreateBaseBodySchema = typebox_1.Type.Object({
    request_url: typebox_1.Type.String({
        description: "The url to associate with the webhook.",
    }),
    name: typebox_1.Type.Optional(typebox_1.Type.String({
        description: "This field will be sent back to you in the notify_url to help match the added account with your user.",
    })),
    format: typebox_1.Type.Optional(ressource_types_js_1.WebhookBodyFormatSchema),
    account_ids: typebox_1.Type.Optional(typebox_1.Type.Array(typebox_1.Type.String(), {
        description: "An optional list of account ids to be targeted by the webhook. If not set, the webhook will apply to all current and future accounts connected to Unipile.",
    })),
    enabled: typebox_1.Type.Optional(typebox_1.Type.Boolean({
        description: "A boolean to activate or disable the webhook.",
    })),
    headers: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookHeaderItemSchema, {
        description: "An optional field to add some headers to the webhook.",
    })),
});
/**
 *
 */
exports.WebhookCreateMessagingBodySchema = typebox_1.Type.Composite([
    exports.WebhookCreateBaseBodySchema,
    typebox_1.Type.Object({
        source: typebox_1.Type.Literal("messaging", {
            description: "A litteral to pick the kind of data you expect to receive from the webhook.",
        }),
        events: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookMessagingTriggerEventSchema, {
            description: 'An array of events on which the webhook should be triggered. Default value include "message_received".',
        })),
        // events: Type.Optional(
        //   Type.Array(
        //     Type.Union([
        //       Type.Literal("message_received"),
        //       Type.Literal("message_read"),
        //       Type.Literal("message_reaction"),
        //     ]),
        //     {
        //       description:
        //         'An array of events on which the webhook should be triggered. Default value include "message_received".',
        //     }
        //   )
        // ),
        data: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookDataItemSchema)),
    }),
], { title: "Messaging webhook" });
/**
 *
 */
exports.WebhookCreateEmailBodySchema = typebox_1.Type.Composite([
    exports.WebhookCreateBaseBodySchema,
    typebox_1.Type.Object({
        source: typebox_1.Type.Literal("email", {
            description: "A litteral to pick the kind of data you expect to receive from the webhook.",
        }),
        events: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookMailTriggerEventSchema, {
            description: 'An array of events on which the webhook should be triggered. Default value include "mail_received".',
        })),
        data: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookEmailDataItemSchema)),
    }),
], { title: "Email webhook" });
/**
 *
 */
exports.WebhookCreateEmailTrackingBodySchema = typebox_1.Type.Composite([
    exports.WebhookCreateBaseBodySchema,
    typebox_1.Type.Object({
        source: typebox_1.Type.Literal("email_tracking", {
            description: "A litteral to pick the kind of data you expect to receive from the webhook.",
        }),
        events: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookMailTrackingTriggerEventSchema, {
            description: 'An array of events on which the webhook should be triggered. Default value include "mail_opened".',
        })),
        data: typebox_1.Type.Optional(typebox_1.Type.Array(ressource_types_js_1.WebhookEmailTrackingDataItemSchema)),
    }),
], { title: "Email Tracking webhook" });
/**
 *
 */
exports.WebhookCreateAccountsStatusBodySchema = typebox_1.Type.Composite([
    exports.WebhookCreateBaseBodySchema,
    typebox_1.Type.Object({
        source: typebox_1.Type.Literal("account_status", {
            description: "A litteral to pick the kind of data you expect to receive from the webhook.",
        }),
    }),
], { title: "Account status webhook" });
/**
 *
 */
exports.WebhookCreateBodySchema = typebox_1.Type.Union([
    exports.WebhookCreateMessagingBodySchema,
    exports.WebhookCreateAccountsStatusBodySchema,
    exports.WebhookCreateEmailBodySchema,
    exports.WebhookCreateEmailTrackingBodySchema,
]);
/**
 *
 */
exports.WebhookCreateBodyValidator = compiler_1.TypeCompiler.Compile(exports.WebhookCreateBodySchema);
// /**
//  *
//  */
// export const getWebhookCreateBodyOpenApiSchema = makeOpenApiSchemaGetter(
//   WebhookCreateBodySchema
// );
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
/**
 *
 */
exports.WebhookCreateResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("WebhookCreated"),
    webhook_id: typebox_1.Type.String(),
});
// /**
//  *
//  */
// export const getWebhookCreateResponseOpenApiSchema = makeOpenApiSchemaGetter(
//   WebhookCreateResponseSchema
// );
/**  */
exports.WebhookCreateResponseValidator = compiler_1.TypeCompiler.Compile(exports.WebhookCreateResponseSchema);
