// unipile-node-sdk/dist/cjs/webhooks/webhooks-list.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookListResponseValidator = exports.WebhookListResponseSchema = exports.WebhookListQueryValidator = exports.WebhookListQuerySchema = exports.WebhookListBaseQuerySchema = exports.WebhookListDecodedCursorValidator = exports.WebhookListDecodedCursorSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const ressource_types_js_1 = require("./ressource.types.js");
const compiler_1 = require("@sinclair/typebox/compiler");
const common_types_js_1 = require("../common/common.types.js");
const query_parameters_type_js_1 = require("../common/query-parameters.type.js");
const query_cursor_js_1 = require("../common/query-cursor.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
/**
 *
 */
exports.WebhookListDecodedCursorSchema = typebox_1.Type.Composite([
    typebox_1.Type.Required(query_parameters_type_js_1.ListLimitQuerySchema),
    typebox_1.Type.Object({
        cursor: typebox_1.Type.Object({
            last_id: common_types_js_1.UniqueIdSchema,
        }),
    }),
]);
exports.WebhookListDecodedCursorValidator = compiler_1.TypeCompiler.Compile(exports.WebhookListDecodedCursorSchema);
/**
 *
 */
exports.WebhookListBaseQuerySchema = query_parameters_type_js_1.ListLimitQuerySchema;
/**
 *
 */
exports.WebhookListQuerySchema = typebox_1.Type.Union([exports.WebhookListBaseQuerySchema, query_parameters_type_js_1.ListCursorQuerySchema]);
/**
 *
 */
exports.WebhookListQueryValidator = compiler_1.TypeCompiler.Compile(exports.WebhookListQuerySchema);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
/**
 *
 */
exports.WebhookListResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('WebhookList'),
    items: typebox_1.Type.Array(typebox_1.Type.Union([
        typebox_1.Type.Composite([typebox_1.Type.Object({ object: typebox_1.Type.Literal('Webhook') }), ressource_types_js_1.WebhookAutoSchema]),
        typebox_1.Type.Composite([typebox_1.Type.Object({ object: typebox_1.Type.Literal('Webhook') }), ressource_types_js_1.WebhookTriggerSchema]),
    ])),
    cursor: typebox_1.Type.Union([(0, query_cursor_js_1.EncodedQueryCursorType)(), typebox_1.Type.Null()]),
}, {
    description: '@todo',
});
/**  */
exports.WebhookListResponseValidator = compiler_1.TypeCompiler.Compile(exports.WebhookListResponseSchema);
