// unipile-node-sdk/dist/esm/posts/comment-send.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentPostResponseValidator = exports.CommentPostResponseSchema = exports.commentPostBodyValidator = exports.CommentPostBodySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const LINKEDIN_COMMENT_CHARACTER_LIMIT = 1250;
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
exports.CommentPostBodySchema = typebox_1.Type.Object({
    account_id: typebox_1.Type.String({
        description: "The id of the account to perform the request from.",
        minLength: 1,
    }),
    text: typebox_1.Type.String({
        minLength: 1,
        maxLength: LINKEDIN_COMMENT_CHARACTER_LIMIT,
    }),
});
exports.commentPostBodyValidator = compiler_1.TypeCompiler.Compile(exports.CommentPostBodySchema);
// export const commentPostBodyOpenApiSchema = makeOpenApiSchemaGetter(
//   CommentPostBodySchema
// );
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.CommentPostResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("CommentSent"),
});
// /**
//  *
//  */
// export const getCommentSentResponseOpenApiSchema = makeOpenApiSchemaGetter(
//   CommentPostResponseSchema
// );
/**  */
exports.CommentPostResponseValidator = compiler_1.TypeCompiler.Compile(exports.CommentPostResponseSchema);
