// unipile-node-sdk/dist/esm/posts/resource.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkedinPostReactionSchema = exports.CommentSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const common_types_js_1 = require("../common/common.types.js");
exports.CommentSchema = typebox_1.Type.Object({
    id: common_types_js_1.UniqueIdSchema,
    author: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
    author_details: typebox_1.Type.Object({
        id: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
        headline: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
        profile_url: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
        network_distance: typebox_1.Type.Union([
            typebox_1.Type.Literal("FIRST_DEGREE"),
            typebox_1.Type.Literal("SECOND_DEGREE"),
            typebox_1.Type.Literal("THIRD_DEGREE"),
            typebox_1.Type.Literal("OUT_OF_NETWORK"),
            typebox_1.Type.Null(),
        ]),
    }),
    date: typebox_1.Type.String(),
    text: typebox_1.Type.String(),
    reaction_counter: typebox_1.Type.Number(),
    reply_counter: typebox_1.Type.Number(),
});
exports.LinkedinPostReactionSchema = typebox_1.Type.Object({
    value: typebox_1.Type.Union([
        typebox_1.Type.Literal("LIKE"),
        typebox_1.Type.Literal("PRAISE"),
        typebox_1.Type.Literal("APPRECIATION"),
        typebox_1.Type.Literal("EMPATHY"),
        typebox_1.Type.Literal("INTEREST"),
        typebox_1.Type.Literal("ENTERTAINMENT"),
    ]),
    author: typebox_1.Type.Object({
        id: common_types_js_1.UniqueIdSchema,
        type: typebox_1.Type.Union([typebox_1.Type.Literal("INDIVIDUAL"), typebox_1.Type.Literal("COMPANY")]),
        name: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
        headline: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
        profile_url: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
        network_distance: typebox_1.Type.Optional(typebox_1.Type.Union([
            typebox_1.Type.Literal("FIRST_DEGREE"),
            typebox_1.Type.Literal("SECOND_DEGREE"),
            typebox_1.Type.Literal("THIRD_DEGREE"),
            typebox_1.Type.Literal("OUT_OF_NETWORK"),
        ])),
    }),
});
