// unipile-node-sdk/dist/esm/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnipileClient = void 0;
const request_sender_js_1 = require("./request-sender.js");
const account_resource_js_1 = require("./resources/account.resource.js");
const email_resource_js_1 = require("./resources/email.resource.js");
const messaging_resource_js_1 = require("./resources/messaging.resource.js");
const users_resource_js_1 = require("./resources/users.resource.js");
const webhook_resource_js_1 = require("./resources/webhook.resource.js");
const state_js_1 = require("./state.js");
class UnipileClient {
    constructor(baseUrl, token, options) {
        //resources
        this.account = new account_resource_js_1.AccountResource(this);
        this.messaging = new messaging_resource_js_1.MessagingResource(this);
        this.users = new users_resource_js_1.UsersResource(this);
        this.email = new email_resource_js_1.EmailResource(this);
        this.webhook = new webhook_resource_js_1.WebhookResource(this);
        this.state = new state_js_1.ClientState(baseUrl, token, options !== null && options !== void 0 ? options : {});
        this.request = new request_sender_js_1.RequestSender(this.state);
    }
}
exports.UnipileClient = UnipileClient;
