// unipile-node-sdk/dist/esm/users/account-owner-profile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountOwnerProfileApiResponseValidator = exports.AccountOwnerProfileApiResponseSchema = exports.ImapAccountOwnerProfileApiResponseSchema = exports.OutlookAccountOwnerProfileApiResponseSchema = exports.GmailAccountOwnerProfileApiResponseSchema = exports.TwitterAccountOwnerProfileApiResponseSchema = exports.TelegramAccountOwnerProfileApiResponseSchema = exports.LinkedinAccountOwnerProfileApiResponseSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const ressource_types_js_1 = require("./ressource.types.js");
const compiler_1 = require("@sinclair/typebox/compiler");
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.LinkedinAccountOwnerProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.LinkedinAccountOwnerProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("AccountOwnerProfile"),
    }),
], { title: "LinkedIn" });
exports.TelegramAccountOwnerProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.TelegramUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("AccountOwnerProfile"),
    }),
], { title: "Telegram" });
exports.TwitterAccountOwnerProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.TwitterUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("AccountOwnerProfile"),
    }),
], { title: "Twitter" });
exports.GmailAccountOwnerProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.GmailUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("AccountOwnerProfile"),
    }),
], { title: "Gmail" });
exports.OutlookAccountOwnerProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.OutlookUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("AccountOwnerProfile"),
    }),
], { title: "Outlook" });
exports.ImapAccountOwnerProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.ImapUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("AccountOwnerProfile"),
    }),
], { title: "Imap" });
/**
 *
 */
exports.AccountOwnerProfileApiResponseSchema = typebox_1.Type.Union([
    exports.LinkedinAccountOwnerProfileApiResponseSchema,
    exports.TelegramAccountOwnerProfileApiResponseSchema,
    exports.TwitterAccountOwnerProfileApiResponseSchema,
    exports.GmailAccountOwnerProfileApiResponseSchema,
    exports.OutlookAccountOwnerProfileApiResponseSchema,
    exports.ImapAccountOwnerProfileApiResponseSchema,
]);
/**  */
exports.AccountOwnerProfileApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.AccountOwnerProfileApiResponseSchema);
