// unipile-node-sdk/dist/esm/users/user-relations-list.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserRelationsListApiResponseValidator = exports.UserRelationsListQueryValidator = exports.UserRelationsListDecodedCursorValidator = exports.UserRelationsListQuerySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const query_cursor_js_1 = require("../common/query-cursor.js");
const query_parameters_type_js_1 = require("../common/query-parameters.type.js");
const common_types_js_1 = require("../common/common.types.js");
const ressource_types_js_1 = require("./ressource.types.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
exports.UserRelationsListQuerySchema = typebox_1.Type.Union([
    typebox_1.Type.Composite([
        query_parameters_type_js_1.LargeListLimitQuerySchema,
        typebox_1.Type.Object({ account_id: common_types_js_1.UniqueIdSchema }),
    ]),
    typebox_1.Type.Object({ account_id: common_types_js_1.UniqueIdSchema, cursor: (0, query_cursor_js_1.EncodedQueryCursorType)() }),
]);
const UserRelationsListDecodedCursorSchema = typebox_1.Type.Object({
    limit: typebox_1.Type.Number(),
    startIndex: typebox_1.Type.Number(),
});
exports.UserRelationsListDecodedCursorValidator = compiler_1.TypeCompiler.Compile(UserRelationsListDecodedCursorSchema);
exports.UserRelationsListQueryValidator = compiler_1.TypeCompiler.Compile(exports.UserRelationsListQuerySchema);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
const UserRelationsListApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("UserRelationsList"),
    items: typebox_1.Type.Array(typebox_1.Type.Composite([
        typebox_1.Type.Object({ object: typebox_1.Type.Literal("UserRelation") }),
        ressource_types_js_1.LinkedinUserRelationSchema,
    ])),
    cursor: typebox_1.Type.Union([(0, query_cursor_js_1.EncodedQueryCursorType)(), typebox_1.Type.Null()]),
});
/**  */
exports.UserRelationsListApiResponseValidator = compiler_1.TypeCompiler.Compile(UserRelationsListApiResponseSchema);
