// unipile-node-sdk/dist/esm/users/user-profile.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfileApiResponseValidator = exports.UserProfileApiResponseSchema = exports.TwitterProfileApiResponseSchema = exports.TelegramProfileApiResponseSchema = exports.InstagramProfileApiResponseSchema = exports.WhatsappProfileApiResponseSchema = exports.LinkedinProfileApiResponseSchema = exports.UserProfileQueryValidator = exports.LinkedinApiTypeEnum = exports.UserProfileQuerySchema = exports.LinkedinSectionSchema = exports.LinkedinSection = void 0;
const typebox_1 = require("@sinclair/typebox");
const ressource_types_js_1 = require("./ressource.types.js");
const common_types_js_1 = require("../common/common.types.js");
const compiler_1 = require("@sinclair/typebox/compiler");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
exports.LinkedinSection = (0, common_types_js_1.StringEnum)(["experience", "education", "languages", "skills", "certifications", "about"], { description: "A string with a section name." });
exports.LinkedinSectionSchema = typebox_1.Type.Union([
    exports.LinkedinSection,
    typebox_1.Type.Literal("*", {
        description: "A string with * character that stands for all sections.",
    }),
    typebox_1.Type.Array(exports.LinkedinSection, { description: "An array of section names." }),
]);
exports.UserProfileQuerySchema = typebox_1.Type.Object({
    account_id: common_types_js_1.UniqueIdSchema,
    notify: typebox_1.Type.Optional(typebox_1.Type.Boolean({
        description: "Whether the profile visit should be notified to the viewee or not. Default is false.",
    })),
    linkedin_api: typebox_1.Type.Optional((0, common_types_js_1.StringEnum)(["recruiter", "sales_navigator"], {
        description: "The Linkedin API that should be used to get the profile (relative features must be subscribed), if different from classic.",
    })),
    linkedin_sections: typebox_1.Type.Optional(exports.LinkedinSectionSchema),
});
var LinkedinApiTypeEnum;
(function (LinkedinApiTypeEnum) {
    LinkedinApiTypeEnum["RECRUITER"] = "recruiter";
    LinkedinApiTypeEnum["SALES_NAVIGATOR"] = "sales_navigator";
})(LinkedinApiTypeEnum || (exports.LinkedinApiTypeEnum = LinkedinApiTypeEnum = {}));
exports.UserProfileQueryValidator = compiler_1.TypeCompiler.Compile(exports.UserProfileQuerySchema);
// export const getUserProfileQueryOpenApiSchema = makeOpenApiSchemaGetter(
//   UserProfileQuerySchema
// );
// export const getUserProfileLinkedinSectionOpenApiSchema =
//   makeOpenApiSchemaGetter(LinkedinSectionSchema);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.LinkedinProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.LinkedinUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("UserProfile"),
    }),
], { title: "LinkedIn" });
exports.WhatsappProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.WhatsappUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("UserProfile"),
    }),
], { title: "Whatsapp" });
exports.InstagramProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.InstagramUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("UserProfile"),
    }),
], { title: "Instagram" });
exports.TelegramProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.TelegramUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("UserProfile"),
    }),
], { title: "Telegram" });
exports.TwitterProfileApiResponseSchema = typebox_1.Type.Composite([
    ressource_types_js_1.TwitterUserProfileSchema,
    typebox_1.Type.Object({
        object: typebox_1.Type.Literal("UserProfile"),
    }),
], { title: "Twitter" });
/**
 *
 */
exports.UserProfileApiResponseSchema = typebox_1.Type.Union([
    exports.LinkedinProfileApiResponseSchema,
    exports.WhatsappProfileApiResponseSchema,
    exports.InstagramProfileApiResponseSchema,
    exports.TelegramProfileApiResponseSchema,
    exports.TwitterProfileApiResponseSchema,
]);
// export const getUserProfileResponseOpenApiSchema = makeOpenApiSchemaGetter(
//   UserProfileApiResponseSchema
// );
/**  */
exports.UserProfileApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.UserProfileApiResponseSchema);
