// unipile-node-sdk/dist/esm/users/user-invitation-sent-list.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserInvitationSentListApiResponseValidator = exports.UserInvitationSentListDecodedCursorValidator = exports.UserInvitationSentListDecodedCursorSchema = exports.UserInvitationSentListQueryValidator = exports.UserInvitationSentListQuerySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const query_parameters_type_js_1 = require("../common/query-parameters.type.js");
const ressource_types_js_1 = require("./ressource.types.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
exports.UserInvitationSentListQuerySchema = typebox_1.Type.Object({
    account_id: typebox_1.Type.String({
        description: "The id of the account to perform the request from.",
        minLength: 1,
    }),
    limit: typebox_1.Type.Optional(query_parameters_type_js_1.ShortLimitParamSchema),
    cursor: typebox_1.Type.Optional(query_parameters_type_js_1.CursorParamSchema),
});
exports.UserInvitationSentListQueryValidator = compiler_1.TypeCompiler.Compile(exports.UserInvitationSentListQuerySchema);
exports.UserInvitationSentListDecodedCursorSchema = typebox_1.Type.Object({
    limit: typebox_1.Type.Number(),
    cursor: typebox_1.Type.Number(),
});
exports.UserInvitationSentListDecodedCursorValidator = compiler_1.TypeCompiler.Compile(exports.UserInvitationSentListDecodedCursorSchema);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
const UserInvitationSentListApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("InvitationList"),
    items: typebox_1.Type.Array(typebox_1.Type.Composite([
        typebox_1.Type.Object({ object: typebox_1.Type.Literal("InvitationSent") }),
        ressource_types_js_1.InvitationSchema,
    ])),
    cursor: typebox_1.Type.Union([typebox_1.Type.String(), typebox_1.Type.Null()]),
});
/**  */
exports.UserInvitationSentListApiResponseValidator = compiler_1.TypeCompiler.Compile(UserInvitationSentListApiResponseSchema);
