// unipile-node-sdk/dist/cjs/messaging/messaging.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessagingAccountTypeEnum = exports.MessagingAccountTypeSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
exports.MessagingAccountTypeSchema = typebox_1.Type.Union([
    typebox_1.Type.Literal("WHATSAPP"),
    typebox_1.Type.Literal("LINKEDIN"),
    typebox_1.Type.Literal("SLACK"),
    typebox_1.Type.Literal("TWITTER"),
    typebox_1.Type.Literal("MESSENGER"),
    typebox_1.Type.Literal("INSTAGRAM"),
    typebox_1.Type.Literal("TELEGRAM"),
]);
var MessagingAccountTypeEnum;
(function (MessagingAccountTypeEnum) {
    MessagingAccountTypeEnum["WHATSAPP"] = "WHATSAPP";
    MessagingAccountTypeEnum["LINKEDIN"] = "LINKEDIN";
    MessagingAccountTypeEnum["SLACK"] = "SLACK";
    MessagingAccountTypeEnum["TWITTER"] = "TWITTER";
    MessagingAccountTypeEnum["MESSENGER"] = "MESSENGER";
    MessagingAccountTypeEnum["INSTAGRAM"] = "INSTAGRAM";
    MessagingAccountTypeEnum["TELEGRAM"] = "TELEGRAM";
})(MessagingAccountTypeEnum || (exports.MessagingAccountTypeEnum = MessagingAccountTypeEnum = {}));
