// unipile-node-sdk/dist/esm/accounts/accounts-reconnect.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountSolveCheckpointApiResponseValidator = exports.AccountSolveCheckpointApiResponseSchema = exports.AccountReconnectApiResponseValidator = exports.AccountReconnectApiResponseSchema = exports.AccountReconnectedApiResponseSchema = exports.AccountReconnectBodyValidator = exports.AccountReconnectBodySchema = exports.AccountReconnectImapMailBodySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const common_types_js_1 = require("../common/common.types.js");
const accounts_create_types_js_1 = require("./accounts-create.types.js");
const compiler_1 = require("@sinclair/typebox/compiler");
/**
 * Imap Mail body schema
 */
exports.AccountReconnectImapMailBodySchema = typebox_1.Type.Object({
    provider: typebox_1.Type.Union([typebox_1.Type.Literal('MAIL')]),
    imap_password: typebox_1.Type.String(),
    smtp_password: typebox_1.Type.String(),
});
/**
 * @todo all provider should have a specific reconnect schema and not use the create one
 */
exports.AccountReconnectBodySchema = typebox_1.Type.Union([
    accounts_create_types_js_1.AccountCreateLinkedinBodySchema,
    accounts_create_types_js_1.AccountCreateWhatsappBodySchema,
    accounts_create_types_js_1.AccountCreateInstagramBodySchema,
    accounts_create_types_js_1.AccountCreateMessengerBodySchema,
    exports.AccountReconnectImapMailBodySchema,
    typebox_1.Type.Omit(accounts_create_types_js_1.AccountCreateGoogleOauthBodySchema, ['sync_limit']),
    accounts_create_types_js_1.AccountCreateTelegramBodySchema,
    typebox_1.Type.Omit(accounts_create_types_js_1.AccountCreateOutlookBodySchema, ['sync_limit']),
    accounts_create_types_js_1.AccountCreateTwitterBodySchema,
    // AccountCreateImapMailBodySchema,
    // AccountCreateMicrosoftMailBodySchema,
]);
exports.AccountReconnectBodyValidator = compiler_1.TypeCompiler.Compile(exports.AccountReconnectBodySchema);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
/**
 *
 */
exports.AccountReconnectedApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('AccountReconnected'),
    account_id: common_types_js_1.UniqueIdSchema,
});
// /**
//  *
//  */
// export const getAccountReconnectedR esponseOpenApiSchema = makeOpenApiSchemaGetter(AccountReconnectedApiResponseSchema);
/**  */
exports.AccountReconnectApiResponseSchema = typebox_1.Type.Union([
    exports.AccountReconnectedApiResponseSchema,
    accounts_create_types_js_1.AccountCreateCheckpointApiResponseSchema
]);
/**  */
exports.AccountReconnectApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.AccountReconnectApiResponseSchema);
/**  */
exports.AccountSolveCheckpointApiResponseSchema = typebox_1.Type.Union([
    accounts_create_types_js_1.AccountCreatedApiResponseSchema,
    exports.AccountReconnectedApiResponseSchema,
    accounts_create_types_js_1.AccountCreateCheckpointApiResponseSchema
]);
/**  */
exports.AccountSolveCheckpointApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.AccountSolveCheckpointApiResponseSchema);
