// unipile-node-sdk/dist/cjs/users/reaction-add.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddPostReactionResponseValidator = exports.addPostReactionResponseSchema = exports.addPostReactionBodyValidator = exports.addPostReactionBodySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const common_types_js_1 = require("../common/common.types.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
exports.addPostReactionBodySchema = typebox_1.Type.Object({
    account_id: typebox_1.Type.String({
        description: "The id of the account to perform the request from.",
        minLength: 1,
    }),
    post_id: typebox_1.Type.String({
        minLength: 1,
        description: "The id of the post that will receive the reaction.",
    }),
    reaction_type: typebox_1.Type.Optional((0, common_types_js_1.StringEnum)(["like", "celebrate", "support", "love", "insightful", "funny"])),
});
exports.addPostReactionBodyValidator = compiler_1.TypeCompiler.Compile(exports.addPostReactionBodySchema);
// export const addPostReactionBodyOpenApiSchema = makeOpenApiSchemaGetter(
//   addPostReactionBodySchema
// );
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.addPostReactionResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("ReactionAdded"),
});
// /**
//  *
//  */
// export const addPostReactionResponseOpenApiSchema = makeOpenApiSchemaGetter(
//   addPostReactionResponseSchema
// );
/**  */
exports.AddPostReactionResponseValidator = compiler_1.TypeCompiler.Compile(exports.addPostReactionResponseSchema);
