// unipile-node-sdk/dist/cjs/users/user-invitation-cancel.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CancelUserInvitationApiResponseValidator = exports.CancelUserInvitationBodySchema = exports.CancelUserInvitationQueryValidator = exports.CancelUserInvitationQuerySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
exports.CancelUserInvitationQuerySchema = typebox_1.Type.Object({
    account_id: typebox_1.Type.String({
        description: "The id of the account to perform the request from.",
        minLength: 1,
    }),
});
exports.CancelUserInvitationQueryValidator = compiler_1.TypeCompiler.Compile(exports.CancelUserInvitationQuerySchema);
exports.CancelUserInvitationBodySchema = typebox_1.Type.Composite([
    exports.CancelUserInvitationQuerySchema,
    typebox_1.Type.Object({ invitation_id: typebox_1.Type.String() }),
]);
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
const CancelUserInvitationResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("InvitationCanceled"),
});
// export const cancelUserUserInvitationResponseOpenApiSchema =
//   makeOpenApiSchemaGetter(CancelUserInvitationResponseSchema);
/**  */
exports.CancelUserInvitationApiResponseValidator = compiler_1.TypeCompiler.Compile(CancelUserInvitationResponseSchema);
