// unipile-node-sdk/dist/esm/common/date.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HourMinuteTimeType = exports.TzidType = exports.UnixTimeMsType = exports.UTCDateTimeMsSchema = exports.UTCDateTimeMsType = exports.UTCDateTimePattern = exports.UTCDateTimeExample = void 0;
const system_1 = require("@sinclair/typebox/system");
const core_types_tmp_js_1 = require("./core.types.tmp.js");
exports.UTCDateTimeExample = '2025-12-31T23:59:59.999Z';
exports.UTCDateTimePattern = '^[1-2]\\d{3}-[0-1]\\d-[0-3]\\dT\\d{2}:\\d{2}:\\d{2}.\\d{3}Z$';
/**
 *
 */
exports.UTCDateTimeMsType = system_1.TypeSystem.Type('UTCDateTimeMs', (options, value) => (0, core_types_tmp_js_1.isUTCDateTimeMs)(value));
/** */
exports.UTCDateTimeMsSchema = (0, exports.UTCDateTimeMsType)({
    description: 'An ISO 8601 UTC datetime (YYYY-MM-DDTHH:MM:SS.sssZ).',
    example: exports.UTCDateTimeExample,
    pattern: exports.UTCDateTimePattern,
});
/**
 *
 */
exports.UnixTimeMsType = system_1.TypeSystem.Type('UnixTimeMs', (options, value) => (0, core_types_tmp_js_1.isUnixTimeMs)(value));
/**
 *
 */
exports.TzidType = system_1.TypeSystem.Type('Tzid', (options, value) => (0, core_types_tmp_js_1.isTzid)(value));
exports.HourMinuteTimeType = system_1.TypeSystem.Type('HourMinuteTime', (options, value) => (0, core_types_tmp_js_1.isHourMinuteTime)(value));
