// unipile-node-sdk/dist/cjs/accounts/account.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountApiResponseValidator = exports.AccounApiResponseSchema = exports.AccountMessengerApiResponse = exports.AccountInstagramApiResponse = exports.AccountTelegramApiResponse = exports.AccountExchangeApiResponse = exports.AccountTwitterApiResponse = exports.AccountSlackApiResponse = exports.AccountLinkedinApiResponse = exports.AccountWhatsappApiResponse = exports.AccountGoogleCalendarApiResponse = exports.AccountOutlookApiResponse = exports.AccountICloudApiResponse = exports.AccountGoogleApiResponse = exports.AccountMailApiResponse = exports.AccountMobileApiResponse = void 0;
const typebox_1 = require("@sinclair/typebox");
const ressource_types_js_1 = require("./ressource.types.js");
const compiler_1 = require("@sinclair/typebox/compiler");
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.AccountMobileApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.MobileAccountWithSourceStatusSchema,
], { title: "Mobile" });
exports.AccountMailApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.MailAccountWithSourceStatusSchema,
], { title: "Mail" });
exports.AccountGoogleApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.GoogleAccountWithSourceStatusSchema,
], { title: "Google" });
exports.AccountICloudApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.ICloudAccountWithSourceStatusSchema,
], { title: "ICloud" });
exports.AccountOutlookApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.OutlookAccountWithSourceStatusSchema,
], { title: "Outlook" });
exports.AccountGoogleCalendarApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.GoogleCalendarAccountWithSourceStatusSchema,
], { title: "Google Calendar" });
exports.AccountWhatsappApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.WhatsAppAccountWithSourceStatusSchema,
], { title: "Whatsapp" });
exports.AccountLinkedinApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.LinkedInAccountWithSourceStatusSchema,
], { title: "Linkedin" });
exports.AccountSlackApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.SlackAccountWithSourceStatusSchema,
], { title: "Slack" });
exports.AccountTwitterApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.TwitterAccountWithSourceStatusSchema,
], { title: "Twitter" });
exports.AccountExchangeApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.ExchangeAccountWithSourceStatusSchema,
], { title: "Exchange" });
exports.AccountTelegramApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.TelegramAccountWithSourceStatusSchema,
], { title: "Telegram" });
exports.AccountInstagramApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.InstagramAccountWithSourceStatusSchema,
], { title: "Instagram" });
exports.AccountMessengerApiResponse = typebox_1.Type.Composite([
    typebox_1.Type.Object({ object: typebox_1.Type.Literal("Account") }),
    ressource_types_js_1.MessengerAccountWithSourceStatusSchema,
], { title: "Messenger" });
/**
 *
 */
exports.AccounApiResponseSchema = typebox_1.Type.Union([
    exports.AccountMobileApiResponse,
    exports.AccountMailApiResponse,
    exports.AccountGoogleApiResponse,
    exports.AccountICloudApiResponse,
    exports.AccountOutlookApiResponse,
    exports.AccountGoogleCalendarApiResponse,
    exports.AccountWhatsappApiResponse,
    exports.AccountLinkedinApiResponse,
    exports.AccountSlackApiResponse,
    exports.AccountTwitterApiResponse,
    exports.AccountExchangeApiResponse,
    exports.AccountTelegramApiResponse,
    exports.AccountInstagramApiResponse,
    exports.AccountMessengerApiResponse,
]);
/**  */
exports.AccountApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.AccounApiResponseSchema);
