// unipile-node-sdk/dist/esm/schemas/account.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.postQrCodeBasedAccountSchema = exports.deleteAccountResponseSchema = exports.postAccountOutputSchema = exports.postAccountResponseSchema = exports.getAccountSourceStatusResponseSchema = exports.getAccountsResponseSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const utils_schema_js_1 = require("./utils.schema.js");
const account = typebox_1.Type.Object({
    id: typebox_1.Type.String(),
    type: typebox_1.Type.String(),
    name: typebox_1.Type.String(),
    created_at: typebox_1.Type.String(),
});
exports.getAccountsResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('AccountList'),
    items: typebox_1.Type.Array(typebox_1.Type.Composite([
        typebox_1.Type.Object({
            object: typebox_1.Type.Literal('Account'),
        }),
        account,
    ])),
});
exports.getAccountSourceStatusResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('AccountSourceStatus'),
    status: typebox_1.Type.Union([
        typebox_1.Type.Literal('OK'),
        typebox_1.Type.Literal('STOPPED'),
        typebox_1.Type.Literal('ERROR'),
        typebox_1.Type.Literal('CREDENTIALS'),
        typebox_1.Type.Literal('PERMISSIONS'),
        typebox_1.Type.Literal('CONNECTING'),
    ]),
});
const accountCreatedApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('AccountCreated'),
    account_id: typebox_1.Type.String(),
});
const whatsappOnGoingAccountCreatedApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('QrCode'),
    account_id: utils_schema_js_1.uniqueIdSchema,
    qrcode: typebox_1.Type.String(),
});
exports.postAccountResponseSchema = typebox_1.Type.Union([
    accountCreatedApiResponseSchema,
    whatsappOnGoingAccountCreatedApiResponseSchema,
]);
exports.postAccountOutputSchema = typebox_1.Type.Union([
    accountCreatedApiResponseSchema,
    typebox_1.Type.Composite([whatsappOnGoingAccountCreatedApiResponseSchema, typebox_1.Type.Object({ qrcodeImage: typebox_1.Type.String() })]),
]);
exports.deleteAccountResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('AccountDeleted'),
});
exports.postQrCodeBasedAccountSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal('Checkpoint'),
    account_id: typebox_1.Type.String(),
    checkpoint: typebox_1.Type.Object({
        type: typebox_1.Type.Literal('QRCODE'),
        qrcode: typebox_1.Type.String(),
    }),
});
