// unipile-node-sdk/dist/esm/mails/mail.delete.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailDeletedApiResponseValidator = exports.MailDeleteQueryValidator = exports.MailDeleteQuerySchema = exports.MailTrashedApiResponseSchema = exports.MailDeletedApiResponseSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const query_parameters_type_js_1 = require("../common/query-parameters.type.js");
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.MailDeletedApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("EmailDeleted"),
}, { description: "@todo Email has been deleted." });
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
exports.MailTrashedApiResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("EmailTrashed"),
}, { description: "@todo Email has been trashed." });
/**
 *
 */
const MailDeleteOptionsQuerySchema = typebox_1.Type.Object({
    account_id: typebox_1.Type.Optional(query_parameters_type_js_1.AccountIdParamSchema),
});
/**
 *
 */
exports.MailDeleteQuerySchema = MailDeleteOptionsQuerySchema;
/**
 *
 */
exports.MailDeleteQueryValidator = compiler_1.TypeCompiler.Compile(exports.MailDeleteQuerySchema);
/**  */
exports.MailDeletedApiResponseValidator = compiler_1.TypeCompiler.Compile(exports.MailDeletedApiResponseSchema);
