// unipile-node-sdk/dist/esm/hosted/hosted-auth-link.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostedAuthLinkResponseValidator = exports.HostedAuthWatchStatusQueryParamsValidator = exports.HostedAuthWatchStatusSchema = exports.HostedAuthLinkResponseSchema = exports.HostedAuthLinkBodyValidator = exports.HostedAuthLinkBodySchema = exports.HostedAuthReconnectLinkBodySchema = exports.HostedAuthCreateLinkBodySchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const compiler_1 = require("@sinclair/typebox/compiler");
const resource_types_js_1 = require("./resource.types.js");
// --------------------------------------------------------------------------
// REQUEST
// --------------------------------------------------------------------------
/**
 * @todo Add some guidance about sensible expiresOn values or remove it
 *       altogether.
 */
exports.HostedAuthCreateLinkBodySchema = typebox_1.Type.Pick(resource_types_js_1.HostedAuthTokenCreateSchema, [
    "type",
    "providers",
    "expiresOn",
    "name",
    "success_redirect_url",
    "failure_redirect_url",
    "notify_url",
    "api_url",
    "disabled_features",
    "sync_limit",
], {
    title: "Connection link",
});
/** */
exports.HostedAuthReconnectLinkBodySchema = typebox_1.Type.Pick(resource_types_js_1.HostedAuthTokenReconnectSchema, [
    "type",
    "reconnect_account",
    "expiresOn",
    "name",
    "success_redirect_url",
    "failure_redirect_url",
    "notify_url",
    "api_url",
    "disabled_features",
    "sync_limit",
], {
    title: "Reconnection link",
});
/** */
exports.HostedAuthLinkBodySchema = typebox_1.Type.Union([
    exports.HostedAuthCreateLinkBodySchema,
    exports.HostedAuthReconnectLinkBodySchema,
]);
/** */
exports.HostedAuthLinkBodyValidator = compiler_1.TypeCompiler.Compile(exports.HostedAuthLinkBodySchema);
// /** */
// export const getHostedAuthLinkBodyOpenApiSchema = makeOpenApiSchemaGetter(
//   HostedAuthLinkBodySchema
// );
// --------------------------------------------------------------------------
// RESPONSE
// --------------------------------------------------------------------------
/**
 * Global schema for every account when final creation step is done
 */
exports.HostedAuthLinkResponseSchema = typebox_1.Type.Object({
    object: typebox_1.Type.Literal("HostedAuthUrl"),
    url: typebox_1.Type.String({
        description: "A url redirecting to Unipile's hosted authentication for account connection or reconnection.",
    }),
});
// /** */
// export const getHostedAuthLinkResponseOpenApiSchema = makeOpenApiSchemaGetter(
//   HostedAuthLinkResponseSchema
// );
/** */
exports.HostedAuthWatchStatusSchema = typebox_1.Type.Optional(typebox_1.Type.Boolean());
exports.HostedAuthWatchStatusQueryParamsValidator = compiler_1.TypeCompiler.Compile(exports.HostedAuthWatchStatusSchema);
exports.HostedAuthLinkResponseValidator = compiler_1.TypeCompiler.Compile(exports.HostedAuthLinkResponseSchema);
