// unipile-node-sdk/dist/esm/attendees/attendees.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailDraftSenderSchema = exports.MailDraftRecipientSchema = exports.AttendeeSchema = exports.IdentifierTypeSchema = void 0;
const typebox_1 = require("@sinclair/typebox");
const common_types_js_1 = require("../common/common.types.js");
/**
 *
 */
exports.IdentifierTypeSchema = (0, common_types_js_1.StringEnum)([
    'CHAT_ATTENDEE_ID',
    'PHONE_NUMBER',
    'EMAIL_ADDRESS',
    'MESSENGER_ID',
    'MESSENGER_THREAD_ID',
    'TIKTOK_ID',
    'TIKTOK_THREAD_ID',
    'TWITTER_ID',
    'TWITTER_THREAD_ID',
    'INSTAGRAM_ID',
    'INSTAGRAM_THREAD_ID',
    'LINKEDIN_ID',
    'LINKEDIN_THREAD_ID',
    'GROUP_THREAD',
]);
/**
 * @todo Review this partially generated type.
 */
exports.AttendeeSchema = typebox_1.Type.Object({
    display_name: typebox_1.Type.Optional(typebox_1.Type.String()),
    profile_picture: typebox_1.Type.Optional(typebox_1.Type.String()),
    identifier: typebox_1.Type.String(),
    identifier_type: exports.IdentifierTypeSchema,
});
/**
 *
 */
exports.MailDraftRecipientSchema = typebox_1.Type.Object({
    display_name: typebox_1.Type.Optional(typebox_1.Type.String({ description: 'The name of the attendee.' })),
    identifier: typebox_1.Type.String({ description: 'The email address of the attendee.' }),
});
/**
 *
 */
exports.MailDraftSenderSchema = typebox_1.Type.Object({
    display_name: typebox_1.Type.Optional(typebox_1.Type.String({ description: 'The name of the attendee.' })),
    identifier: typebox_1.Type.Optional(typebox_1.Type.String({ description: 'The email address of the attendee.' })),
});
